### PR TITLE
Refactor auditing Cypress tests

### PR DESCRIPTION
**Pull request by nemanjaglumac**
### Status
READY

### What does this PR do?
- Restructures the way auditing tests are organized
- Makes all tests run in isolation (as part of #14339)
- Extracts setup into `before()` hook
- Introduces some of the Cypress official best practices

### How to test this manually?
- Put `.only` to **any** `describe()` block or any test within any `describe()` block
- Test(s) should pass

### CODE REVIEW

File: enterprise/frontend/test/metabase-enterprise/audit/auditing.cy.spec.js [hunk 1]
```diff
@@ -141,29 +114,29 @@ describeWithToken("audit > auditing", () => {
...
+
       cy.get("tr")
         .last()
         .children()
         .eq(-2)
+        .as("lastActive")
         .should("contain", year);
     });
 
     it.skip("should load the Audit log (Audit log should display views of dashboards)", () => {
```
Comment (nemanjaglumac) [inline enterprise/frontend/test/metabase-enterprise/audit/auditing.cy.spec.js:129]: @flamber do you know/remember why this one is skipped?

File: enterprise/frontend/test/metabase-enterprise/audit/auditing.cy.spec.js [hunk 2]
```diff
@@ -13,108 +11,81 @@ import { SAMPLE_DATASET } from "__support__/cypress_sample_dataset";
...
-              display_name: "ID",
-              type: "dimension",
-              dimension: ["field-id", PRODUCTS.ID],
-              "widget-type": "category",
-              default: null,
-            },
+const QUESTION = "question";
+const DASHBOARD = "dashboard";
+
+export function generateQuestions(user) {
```
Comment (nemanjaglumac) [inline enterprise/frontend/test/metabase-enterprise/audit/auditing.cy.spec.js:17]: Just realized these functions were being exported. As soon as CI finishes running, I'll fix this.

File: enterprise/frontend/test/metabase-enterprise/audit/auditing.cy.spec.js (hunk 1 above)
Comment (flamber) [reply enterprise/frontend/test/metabase-enterprise/audit/auditing.cy.spec.js:130]: H2 problems EE#287

**Review by flamber — APPROVED** [review]
`userRoles` would be easier to read if it used keys instead of indexes, but perhaps there's a technical reason for that.
`QUESTION`/`DASHBOARD` variables should probably just be removed and replaced with the text string instead.

**Comment by nemanjaglumac** [comment]
> `userRoles` would be easier to read if it used keys instead of indexes, but perhaps there's a technical reason for that.

I think that decision was made probably because `singIn()` function expects the literal string "admin" or "normal". I could've solved it with:
```javascript
const userRoles = ([admin, normal] = Object.keys(USERS));
```
but it felt strange to have the variable with the same name as the string. Although I could've named them however I wanted, I felt the meaning would've been lost if I change variable names. Does this look better to you @flamber ?

> `QUESTION`/`DASHBOARD` variables should probably just be removed and replaced with the text string instead.

Makes sense. The only reason why I included them was to somehow tie the names in helper functions with the strings that we're asserting later on. It's probably an overkill.

**Comment by flamber** [comment]
This refactor is better than before, but it was one place where I "had to know" instead of just reading the line. It's a small inconvenience. Like with `` cy.log(`**View ${userRoles[1]}'s dashboard**`); ``
I'm okay with userRoles indexes if you're good with it - no hassle.

**Pull request merged**